main.py

```diff
@@ -505,6 +509,18 @@ def has_capability(self, capability):
     def get_capability(self, capability):
         return self.capabilities.get(capability)
 
+    def shutdown(self):
+        self.send_request(Request.shutdown(), self.exit)
```

Comment:
Maybe we should make this a blocking request instead of a non-blocking one?

Reply:
I'd like to understand the case where this is important, and I'd like to know the impact to sublime text / LSP's own process before we do this.